exonum-java-binding-common/src/main/java/com/exonum/binding/common/blockchain/Block.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.blockchain;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.google.auto.value.AutoValue;
+
+/**
+ * Exonum block header data structure.
+ *
+ * A block is essentially a list of transactions, which is a result of the consensus algorithm
+ * (thus authenticated by the supermajority of validators) and is applied atomically to the
+ * blockchain state.
+ *
+ * This structure only contains the amount of transactions and the transactions root hash as well as
+ * other information, but not the transactions themselves.
+ */
+@AutoValue
+public abstract class Block {
+
+  public static Block valueOf(
```

Comment:
We don't have that in core.

Reply:
Yes, but not exactly. The core doesn't have a hash as a stored *property*, but it gives you an instance method `hash` to compute it, because `Block` used to be (can't say about the protobuf implementation) a wrapper around a binary buffer. As `Block` is identified by its hash, I'd include that in the Java `Block`.